tools/gen_runtime_vectors/main.go

```diff
@@ -0,0 +1,359 @@
+								Denomination: "ROSE",
+							},
+						},
+						{
+							types.BaseUnits{
+								Amount:       *quantity.NewFromUint64(0),
+								Denomination: "TEST",
+							},
+						},
+						{},
+					} {
+						for _, id := range []uint64{0, 1, math.MaxUint64} {
+							for _, d := range []map[string]interface{}{
+								// Valid data, empty
+								{},
+								// Valid data, one function call with argument.
+								{
+									"instantiate": map[string]interface{}{
+										"initial_counter": 42,
+									},
+								},
+								// Valid data, one function call with argument.
+								{
+									"say_hello": map[string]interface{}{
+										"who": "me",
+									},
+								},
+								// Valid data, custom ABI.
+								{
+									"test123": "test1234",
+								},
+							} {
+								addr, _ := helpers.ResolveAddress(nil, daveNativeAddr)
+								for _, p := range []contracts.Policy{
+									// Valid policy, everyone can instantiate/upgrade it.
+									{Everyone: &struct{}{}},
+									// Valid policy, noone can instantiate/upgrade it.
+									{Nobody: &struct{}{}},
+									// Valid policy, arbitrary address can instantiate/upgrade it.
+									{Address: addr},
+								} {
+									// contracts.Upload not supported by Ledger due to tx bytecode size.
+
+									// contracts.Instantiate
+									txBodyInstantiate := &contracts.Instantiate{
+										CodeID:         contracts.CodeID(id),
+										UpgradesPolicy: p,
+										Data:           cbor.Marshal(d),
+										Tokens:         tokens,
+									}
+									tx = contracts.NewInstantiateTx(fee, txBodyInstantiate)
+									vectors = append(vectors, MakeRuntimeTestVector(tx, txBodyInstantiate, meta, t.valid, t.signer, nonce, sigCtx))
+								}
+
+								// contracts.Call
+								txBodyCall := &contracts.Call{
+									ID:     contracts.InstanceID(id),
+									Data:   cbor.Marshal(d),
+									Tokens: tokens,
+								}
+								tx = contracts.NewCallTx(fee, txBodyCall)
+								vectors = append(vectors, MakeRuntimeTestVector(tx, txBodyCall, meta, t.valid, t.signer, nonce, sigCtx))
+
+								// contracts.Upgrade
+								txBodyUpgrade := &contracts.Upgrade{
+									ID:     contracts.InstanceID(id),
+									CodeID: contracts.CodeID(0 ^ id),
+									Data:   cbor.Marshal(d),
+									Tokens: tokens,
+								}
+								tx = contracts.NewUpgradeTx(fee, txBodyUpgrade)
+								vectors = append(vectors, MakeRuntimeTestVector(tx, txBodyUpgrade, meta, t.valid, t.signer, nonce, sigCtx))
+							}
+						}
+					}
+
+					// Encrypted transaction, body is types.CallFormatEncryptedX25519DeoxysII.
+					body := &struct {
+						Pk    []byte
+						Nonce []byte
+						Data  []byte
+					}{
+						[]byte{'p', 'u', 'b', 'l', 'i', 'c', 'k', 'e', 'y'},
+						[]byte{'e', 'n', 'c', 'r', 'y', 'p', 't', 'i', 'o', 'n', 'n', 'o', 'n', 'c', 'e'},
+						[]byte("encrypted Call.Body object goes here"),
```

Comment:
Any reason why the first two are written out with individual characters?

Reply:
Some leftovers, sorry.